transcom/mymove
# [ MB-16244 ] Addition of reweighs to query for updated-since params in list prime moves

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16244)

## Summary

In this we include reweighs in the query that limits the returned moves to only include moves updated since a given time parameter. Reweighs are not directly attached to the moves, so we have to do join the reweighs to the shipments.

I'd love someone to double check my sql. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Hit the List Prime Moves endpoint and note that it is returning all prime moves. Find a Move to test that has a shipment that we can request a reweigh. With a fresh dev db, I used RISKEX.
2. Update the `since` parameter to be recent enough to not return any moves. A few seconds ago would make sense. I made a fresh dev db and then copied the `updated_at` time from the RISKEX move and then added a few seconds. ![image](https://github.com/transcom/mymove/assets/110134470/5030fac3-01ec-46ef-b538-a7813063f5c3)

3. Log in to the office UI as a TOO and navigate to the move that you chose in step 1. 
4. Request a Reweigh
![image](https://github.com/transcom/mymove/assets/110134470/1fca94d0-0afb-454c-9264-46d7fa50828d)
5. Refresh the List Prime Moves endpoint with the most recent search parameter and you should see your move as the only move. 
![image](https://github.com/transcom/mymove/assets/110134470/d0fcfbf4-1ea0-4817-9fee-35681903df79)



### Backend



## Screenshots
I included screenshots in the testing instructions.
